scripts/ObstacleDetector.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+import rospy
+import math
+from std_msgs.msg import Bool, Header, String, Float64, Float32
+from sensor_msgs.msg import LaserScan
+import genpy
+import numpy as np
+from geometry_msgs.msg import Point,PointStamped
+class ObstacleDetector():
+    def __init__(self):
+        self.pub0 = rospy.Publisher('/estop', Bool,  queue_size=10) # init your publishers early
+        self.pubObstPoint = rospy.Publisher('/point2follow', PointStamped, queue_size=10)
+        rospy.init_node('ObstacleDetector', anonymous=True)
+        self.subScan = rospy.Subscriber('/front/scan',LaserScan,self.callback)
+        self.rate = rospy.Rate(1)
+        self.scan = None
+
+    def callback(self,data):
+        self.scan = data
+
+    def detectObstacles(self):
+    # Code is supposed to detect if there is an obstacle, and if so, stop the tractor
+    # lines with *** coordinated to the bravobot - 10/15/17
+    # Things do to before completion:
+    # TODO: Fix this up a little bit and make it more robust
+    # Actually import data -- Done
+    # Find out how to stop the tractor -- In Progress
+    # Test various values to make sure it works
+        totalDist = [] #Setting arrays
```

Comment:
Not reviewing this currently, since we need to move on, but it is worth saying that the math needs to be explained more clearly (in documentation or in code). This is not currently maintainable by anyone but the author.